lib/auth/clt_test.go

```diff
@@ -77,3 +86,100 @@ func TestClient_DialTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestClient_RequestTimeout(t *testing.T) {
+	t.Parallel()
+
+	testDone := make(chan struct{})
+	sawRoot := make(chan bool, 1)
+	sawSlow := make(chan bool, 1)
+
+	srv := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/v2/authorities/host/rotate/external" {
+			sawRoot <- true
+			http.Redirect(w, r, "/slow", http.StatusFound)
+			return
+		}
+
+		if r.URL.Path == "/slow" {
+			sawSlow <- true
+			w.Write([]byte("Hello"))
+			w.(http.Flusher).Flush()
+
+			<-testDone
+			return
+		}
+	}))
+	t.Cleanup(srv.Close)
+	t.Cleanup(func() { close(testDone) }) // before srv.Close, to unblock /slow handler
+
+	cfg := apiclient.Config{
+		Addrs: []string{srv.Listener.Addr().String()},
+		Credentials: []apiclient.Credentials{
+			apiclient.LoadTLS(&tls.Config{
+				InsecureSkipVerify: true,
+			}),
+		},
+	}
+	clt, err := NewClient(cfg)
+	require.NoError(t, err)
+
+	require.NotEmpty(t, cfg.Credentials)
+	tlsCfg, err := cfg.Credentials[0].TLSConfig()
+	require.NoError(t, err)
+	srv.TLS = tlsCfg
+
+	srv.StartTLS()
+
+	ca := newCertAuthority(t, "test", types.HostCA)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	t.Cleanup(cancel)
+
+	err = clt.RotateExternalCertAuthority(ctx, ca)
+	require.ErrorIs(t, trace.Unwrap(err), context.DeadlineExceeded)
+
+	select {
+	case <-sawRoot:
+		// good.
+	default:
+		t.Fatal("handler never got /v2/authorities/host/rotate/external request")
+	}
+
+	select {
+	case <-sawSlow:
+		// good.
+	default:
+		t.Fatal("handler never got /slow request")
+	}
```

Comment:
nit: Fun trick :) Closed/empty channels return the default value.
```suggestion
	close(sawRoot)
	require.True(t, <-sawRoot, "handler never got /v2/authorities/host/rotate/external request")
	
	close(sawSlow)
	require.True(t, <-sawSlow, "handler never got /slow request")
```